### PR TITLE
Feature/vertical tab

### PR DESCRIPTION
**Pull request by EdwardIrby**
Questions:
 - Should users of component manually have to manage active syncing between tier 1 and tier 2

### CODE REVIEW

**Comment by EdwardIrby** [comment]
@sweeting is there a max or min width on vertical tabs?

File: packages/verticaltabs/src/react/__stories__/index.story.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { storiesOf } from '@storybook/react'
+import React from 'react'
+
+import VerticalTabs from '../index.js'
+
+storiesOf('Vetical Tabs', module)
+  .addDecorator(story => <div style={{ height: '100vh' }}>{story()}</div>)
+  .add('Group', _ => (
+    <VerticalTabs>
```
Comment (danethurber) [inline packages/verticaltabs/src/react/__stories__/index.story.js]: in this example both the Group header and Tier2 header are `h2`. we need to device of a way where that doesn't happen - maybe it's configurable and prop to pass in the headerTag.

another option is more compound components :) - i could see something like this being more flexible

```
<VerticalTabs.Group>
  <VerticalTabs.Group.Header tag="h2">Tab Group</VerticalTabs.Group.Header>

  <VerticalTabs.List>
    <VerticalTabs.List.Header tag="h3">Tier 1 a</VerticalTabs.List.Header>

    <VerticalTabs.List.Item>Tier 2 a</VerticalTabs.List.Header>
    <VerticalTabs.List.Item>Tier 2 b</VerticalTabs.List.Header>

  </VerticalTabs.List>
```
another option is passing element as props

```
<VerticalTabs.Group heading={<VerticalTabs.Header tag="h2">Tab Group</VerticalTabs.Header>}>
...
</VerticalTabs.Group>
```

Comment (danethurber) [reply packages/verticaltabs/src/react/__stories__/index.story.js]: another concern is the the group heading needs to be able to have a data attrs on it for gitprime to use their onboarding tool. a more explicit compound component would add that flexibility. let me know if you want to talk about different approaches

**Comment by bressain** [comment]
I don't think I could add anything useful here. Should probably remove me as a reviewer.

File: packages/verticaltabs/src/css/index.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+import core from '@pluralsight/ps-design-system-core'
+import { compose, css, media } from 'glamor'
+
+
+export default {
+  '.psds-verticaltabs':{
+    width: '240px',
+    height: '100%',
+    overflowY: 'auto',
+    padding: `${core.layout.spacingMedium} 0`,
```
[the remark this answers is not in the document]
Comment (EdwardIrby) [reply packages/verticaltabs/src/css/index.js]: @sweeting We're not including the collapse right? So the 40px space above should we include that?

File: packages/verticaltabs/src/css/index.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  },
+  '.psds-verticaltabs__item-icon': {
+    marginRight: core.layout.spacingSmall,
+  },
+  '.psds-verticaltabs__item-link': {
+    textDecoration: 'none',
+    color: 'inherit',
+    font: 'inherit',
+  },
+  '.psds-verticaltabs__button': {
```
[the remark this answers is not in the document]
Comment (EdwardIrby) [reply packages/verticaltabs/src/css/index.js]: generic button styles

File: packages/verticaltabs/src/css/index.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    backgroundColor: core.colors.gray05
+  },
+  '.psds-verticaltabs__divider': {
+    margin: `${core.layout.spacingXSmall} 0`,
+  },
+  [`.psds-verticaltabs__divider.psds-theme--${themeNames.light}`]: {
+    borderTop: `1px solid ${core.colors.gray01}`,
+  },
+  [`.psds-verticaltabs__divider.psds-theme--${themeDefaultName}`]: {
+    borderTop: `1px solid ${core.colors.gray03}`,
```
[the remark this answers is not in the document]
Comment (EdwardIrby) [reply packages/verticaltabs/src/css/index.js]: need to fix this for bottom and remove inset

File: packages/verticaltabs/src/css/index.js [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+  '.psds-verticaltabs__list': {
+    listStyle: 'none',
+    margin: 0,
+    padding: 0,
+  },
+  '.psds-verticaltabs__group-collapsible-list': {
+    overflow: 'hidden',
+    visibility: 'hidden',
+    transition: `height ${core.motion.speedNormal}`
+  }
```
[the remark this answers is not in the document]
Comment (EdwardIrby) [reply packages/verticaltabs/src/css/index.js]: No I coded the margin on the header but Have switch it to just set the header height to 32 like in the comp.

File: packages/site/pages/components/verticaltabs.js [hunk 6]
```diff
@@ -0,0 +1,399 @@
...
+  PageHeading,
+  PropTypes,
+  SectionHeading,
+  TextLink,
+  ThemeToggle
+} from '../../src/ui/index.js'
+
+export default _ => (
+  <Chrome>
+    <Content title="Tab">
```
Comment (danethurber) [inline packages/site/pages/components/verticaltabs.js]:
```suggestion
    <Content title="VerticalTabs">
```

File: packages/site/pages/components/verticaltabs.js [hunk 7]
```diff
@@ -0,0 +1,399 @@
...
+      <PageHeading packageName="tab">Vertical Tabs</PageHeading>
```
Comment (danethurber) [inline packages/site/pages/components/verticaltabs.js]:
```suggestion
      <PageHeading packageName="verticaltabs">Vertical Tabs</PageHeading>
```

File: packages/site/pages/components/verticaltabs.js [hunk 8]
```diff
@@ -0,0 +1,399 @@
...
+
+      <P>Install the component dependency:</P>
+      <Code language="bash">
+        npm install @pluralsight/ps-design-system-verticaltabs
+      </Code>
+
+      <P>Include a React component in your project:</P>
+      <Code language="javascript">
+        import Tab from '@pluralsight/ps-design-system-verticaltabs/react'
```
Comment (danethurber) [inline packages/site/pages/components/verticaltabs.js]:
```suggestion
        import VerticalTabs from '@pluralsight/ps-design-system-verticaltabs/react'
```

File: packages/verticaltabs/package.json [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    "test": "jest",
+    "test:watch": "npm run test -- --watchAll"
+  },
+  "dependencies": {
+    "@pluralsight/ps-design-system-collapsible": "^1.0.0",
+    "@pluralsight/ps-design-system-core": "^4.4.5",
+    "@pluralsight/ps-design-system-filter-react-props": "*",
+    "@pluralsight/ps-design-system-icon": "^13.0.1",
+    "@pluralsight/ps-design-system-prop-types": "^2.0.7",
+    "@pluralsight/ps-design-system-theme": "*",
```
Comment (danethurber) [inline packages/verticaltabs/package.json]:
```suggestion
```

File: packages/verticaltabs/src/react/index.js [hunk 10]
```diff
@@ -0,0 +1,330 @@
...
+  ]),
+  icon: PropTypes.node
+}
+Tier2.Header = forwardRef(({ active, children, ...rest }, ref) => {
+  const TagName = rest.href ? 'a' : rest.onClick ? 'button' : 'span'
+  return (
+    <TagName
+      {...styles.itemTag(TagName)}
+      {...styles.tier2Header()}
+      {...filterReactProps(rest, { tagName: 'tier2' })}
```
Comment (danethurber) [inline packages/verticaltabs/src/react/index.js]:
```suggestion
      {...filterReactProps(rest, { tagName: TagName })}
```

File: packages/verticaltabs/src/react/index.js [hunk 11]
```diff
@@ -0,0 +1,330 @@
...
+}
+Divider.displayName = 'VerticalTabs.Divider'
+
+const VerticalTabs = forwardRef(({ children, ...rest }, ref) => {
+  const themeName = useTheme()
+  return (
+    <ul
+      ref={ref}
+      {...styles.verticaltabs(themeName)}
+      {...filterReactProps(rest)}
```
Comment (danethurber) [inline packages/verticaltabs/src/react/index.js]:
```suggestion
      {...filterReactProps(rest, { tagName: 'ul' })}
```

**Pull request merged**